server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4554,6 +4581,112 @@ public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean re
+        try {
+            String segmentInfosSnapshotFilename = null;
+            String segmentInfosSnapshotPrefix = syncTillCommitPoint ? IndexFileNames.SEGMENTS : SEGMENT_INFO_SNAPSHOT_FILENAME_PREFIX;
+            Set<String> localSegmentFiles = Sets.newHashSet(storeDirectory.listAll());
+            for (String file : uploadedSegments.keySet()) {
+                long checksum = Long.parseLong(uploadedSegments.get(file).getChecksum());
+
+                if (overrideLocal || localDirectoryContains(storeDirectory, file, checksum) == false) {
+                    if (localSegmentFiles.contains(file)) {
+                        storeDirectory.deleteFile(file);
+                    }
+                    if (remoteSegmentStoreDirectory != null) {
+                        String uploadedFileName = uploadedSegments.get(file).toString().split("::")[1];
+                        logger.debug(
+                            "local file is "
+                                + file
+                                + "uploaded file name is "
+                                + uploadedFileName
+                                + " and file is "
+                                + uploadedSegments.get(file).toString()
+                        );
+                        remoteSegmentStoreDirectory.copyFrom(
+                            tempRemoteSegmentDirectory,
+                            file,
+                            file,
+                            IOContext.DEFAULT,
+                            false,
+                            uploadedSegments.get(file).getChecksum()
+                        );
+                    }
+                    storeDirectory.copyFrom(tempRemoteSegmentDirectory, file, file, IOContext.DEFAULT);
+                    downloadedSegments.add(file);
+                    if (file.startsWith(segmentInfosSnapshotPrefix)) {
+                        assert segmentInfosSnapshotFilename == null : "There should be only one SegmentInfosSnapshot file";
+                        segmentInfosSnapshotFilename = file;
+                    }
+                } else {
+                    logger.info("file " + file + " is not able to download, adding to skipped list");
+                    skippedSegments.add(file);
+                }
+            }
+
+            if (segmentInfosSnapshotFilename != null) {
+                if (remoteSegmentStoreDirectory != null) {
+                    remoteSegmentStoreDirectory.uploadMetadata(uploadedSegments.keySet(), storeDirectory, primaryTerm, commitGeneration);
+                }
+                try (
+                    ChecksumIndexInput indexInput = new BufferedChecksumIndexInput(
+                        storeDirectory.openInput(segmentInfosSnapshotFilename, IOContext.DEFAULT)
+                    )
+                ) {
+                    SegmentInfos infosSnapshot = SegmentInfos.readCommit(store.directory(), indexInput, commitGeneration);
```

Comment:
We don't need to explicitly read `segments_N` file. This code is required only for snapshot file (to get the refresh level durability)

Reply:
ack, removed this code